core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -150,29 +150,46 @@ public static void callAuthorizationPlugin(
       if (needApplyAllAuthorizationPlugin(securableObject)) {
         Catalog[] catalogs = catalogManager.listCatalogsInfo(Namespace.of(metalake));
         for (Catalog catalog : catalogs) {
-          callAuthorizationPluginImpl(catalogsAlreadySet, consumer, catalog);
+          callAuthorizationPluginImpl(consumer, catalog);
         }
 
       } else if (supportsSingleAuthorizationPlugin(securableObject.type())) {
         NameIdentifier catalogIdent =
             NameIdentifierUtil.getCatalogIdentifier(
                 MetadataObjectUtil.toEntityIdent(metalake, securableObject));
         Catalog catalog = catalogManager.loadCatalog(catalogIdent);
-        callAuthorizationPluginImpl(catalogsAlreadySet, consumer, catalog);
+        if (!catalogsAlreadySet.contains(catalog.name())) {
+          catalogsAlreadySet.add(catalog.name());
+          callAuthorizationPluginImpl(consumer, catalog);
+        }
+      }
+    }
+  }
+
+  public static void callAuthorizationPluginForMetadataObject(
+      String metalake, MetadataObject metadataObject, Consumer<AuthorizationPlugin> consumer) {
+    CatalogManager catalogManager = GravitinoEnv.getInstance().catalogManager();
+    if (needApplyAuthorizationPluginSetOwner(metadataObject.type())) {
```

Comment:
I think we can directly judge `if (metadataObject.type() == MetadataObject.Type.METALAKE)` would be clear.

Reply:
I changed to `needApplyAllAuthorizationPlugin`